src/IterableWeakMap.ts

```diff
@@ -0,0 +1,110 @@
+interface WakMapValue<K extends object, V> {
+    value: V
+    ref: WeakRef<K>
+}
+
+export class IterableWeakMap<K extends object = object, V = any> implements Map<K, V> {
+    private readonly weakMap = new WeakMap<K, WakMapValue<K, V>>()
+    private readonly refSet: Set<WeakRef<K>> = new Set()
+    private readonly finalizationGroup = new FinalizationRegistry(IterableWeakMap.cleanup)
```

Comment:
Let's add fallback for non weakRef/map in case we run in a non supporting browser.
We'll have a memory leak in those browsers, but at least let's not explode with runtime exception (maybe add a console warning).